### ISSUE TITLE
[camerax] camera_android_camerax doesn't support wide field of view on Galaxy XCover6 Pro (SM-G736B) 

### ISSUE THREAD

**Issue by livnehra** [flutter/flutter#146427]
### What package does this bug report belong to?

camera

### What target platforms are you seeing this bug on?

Android

### Have you already upgraded your packages?

Yes

### Dependency versions

<details><summary>pubspec.lock</summary>

```lock
# Generated by pub
# See https://dart.dev/tools/pub/glossary#lockfile
packages:
  async:
    dependency: transitive
    description:
      name: async
      sha256: "947bfcf187f74dbc5e146c9eb9c0f10c9f8b30743e341481c1e2ed3ecc18c20c"
      url: "https://pub.dev"
    source: hosted
    version: "2.11.0"
  boolean_selector:
    dependency: transitive
    description:
      name: boolean_selector
      sha256: "6cfb5af12253eaf2b368f07bacc5a80d1301a071c73360d746b7f2e32d762c66"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.1"
  camera:
    dependency: "direct main"
    description:
      name: camera
      sha256: "9499cbc2e51d8eb0beadc158b288380037618ce4e30c9acbc4fae1ac3ecb5797"
      url: "https://pub.dev"
    source: hosted
    version: "0.10.5+9"
  camera_android:
    dependency: transitive
    description:
      name: camera_android
      sha256: ae5b9a996dfb8d77b02031b67f5500873d6402f33bd6a5283e932eef08542a51
      url: "https://pub.dev"
    source: hosted
    version: "0.10.9"
  camera_android_camerax:
    dependency: "direct main"
    description:
      name: camera_android_camerax
      sha256: "6a67149f9b05119eb9d047ff9f79280288c88690c390a6a3b5f0993bc7ccb39e"
      url: "https://pub.dev"
    source: hosted
    version: "0.6.2"
  camera_avfoundation:
    dependency: transitive
    description:
      name: camera_avfoundation
      sha256: "608b56b0880722f703871329c4d7d4c2f379c8e2936940851df7fc041abc6f51"
      url: "https://pub.dev"
    source: hosted
    version: "0.9.13+10"
  camera_platform_interface:
    dependency: transitive
    description:
      name: camera_platform_interface
      sha256: a250314a48ea337b35909a4c9d5416a208d736dcb01d0b02c6af122be66660b0
      url: "https://pub.dev"
    source: hosted
    version: "2.7.4"
  camera_web:
    dependency: transitive
    description:
      name: camera_web
      sha256: "9e9aba2fbab77ce2472924196ff8ac4dd8f9126c4f9a3096171cd1d870d6b26c"
      url: "https://pub.dev"
    source: hosted
    version: "0.3.3"
  characters:
    dependency: transitive
    description:
      name: characters
      sha256: "04a925763edad70e8443c99234dc3328f442e811f1d8fd1a72f1c8ad0f69a605"
      url: "https://pub.dev"
    source: hosted
    version: "1.3.0"
  clock:
    dependency: transitive
    description:
      name: clock
      sha256: cb6d7f03e1de671e34607e909a7213e31d7752be4fb66a86d29fe1eb14bfb5cf
      url: "https://pub.dev"
    source: hosted
    version: "1.1.1"
  collection:
    dependency: transitive
    description:
      name: collection
      sha256: f092b211a4319e98e5ff58223576de6c2803db36221657b46c82574721240687
      url: "https://pub.dev"
    source: hosted
    version: "1.17.2"
  cross_file:
    dependency: transitive
    description:
      name: cross_file
      sha256: "2f9d2cbccb76127ba28528cb3ae2c2326a122446a83de5a056aaa3880d3882c5"
      url: "https://pub.dev"
    source: hosted
    version: "0.3.3+7"
  cupertino_icons:
    dependency: "direct main"
    description:
      name: cupertino_icons
      sha256: d57953e10f9f8327ce64a508a355f0b1ec902193f66288e8cb5070e7c47eeb2d
      url: "https://pub.dev"
    source: hosted
    version: "1.0.6"
  fake_async:
    dependency: transitive
    description:
      name: fake_async
      sha256: "511392330127add0b769b75a987850d136345d9227c6b94c96a04cf4a391bf78"
      url: "https://pub.dev"
    source: hosted
    version: "1.3.1"
  flutter:
    dependency: "direct main"
    description: flutter
    source: sdk
    version: "0.0.0"
  flutter_lints:
    dependency: "direct dev"
    description:
      name: flutter_lints
      sha256: a25a15ebbdfc33ab1cd26c63a6ee519df92338a9c10f122adda92938253bef04
      url: "https://pub.dev"
    source: hosted
    version: "2.0.3"
  flutter_plugin_android_lifecycle:
    dependency: transitive
    description:
      name: flutter_plugin_android_lifecycle
      sha256: b068ffc46f82a55844acfa4fdbb61fad72fa2aef0905548419d97f0f95c456da
      url: "https://pub.dev"
    source: hosted
    version: "2.0.17"
  flutter_test:
    dependency: "direct dev"
    description: flutter
    source: sdk
    version: "0.0.0"
  flutter_web_plugins:
    dependency: transitive
    description: flutter
    source: sdk
    version: "0.0.0"
  js:
    dependency: transitive
    description:
      name: js
      sha256: f2c445dce49627136094980615a031419f7f3eb393237e4ecd97ac15dea343f3
      url: "https://pub.dev"
    source: hosted
    version: "0.6.7"
  lints:
    dependency: transitive
    description:
      name: lints
      sha256: "0a217c6c989d21039f1498c3ed9f3ed71b354e69873f13a8dfc3c9fe76f1b452"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.1"
  matcher:
    dependency: transitive
    description:
      name: matcher
      sha256: "1803e76e6653768d64ed8ff2e1e67bea3ad4b923eb5c56a295c3e634bad5960e"
      url: "https://pub.dev"
    source: hosted
    version: "0.12.16"
  material_color_utilities:
    dependency: transitive
    description:
      name: material_color_utilities
      sha256: "9528f2f296073ff54cb9fee677df673ace1218163c3bc7628093e7eed5203d41"
      url: "https://pub.dev"
    source: hosted
    version: "0.5.0"
  meta:
    dependency: transitive
    description:
      name: meta
      sha256: "3c74dbf8763d36539f114c799d8a2d87343b5067e9d796ca22b5eb8437090ee3"
      url: "https://pub.dev"
    source: hosted
    version: "1.9.1"
  path:
    dependency: transitive
    description:
      name: path
      sha256: "8829d8a55c13fc0e37127c29fedf290c102f4e40ae94ada574091fe0ff96c917"
      url: "https://pub.dev"
    source: hosted
    version: "1.8.3"
  plugin_platform_interface:
    dependency: transitive
    description:
      name: plugin_platform_interface
      sha256: "4820fbfdb9478b1ebae27888254d445073732dae3d6ea81f0b7e06d5dedc3f02"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.8"
  sky_engine:
    dependency: transitive
    description: flutter
    source: sdk
    version: "0.0.99"
  source_span:
    dependency: transitive
    description:
      name: source_span
      sha256: "53e943d4206a5e30df338fd4c6e7a077e02254531b138a15aec3bd143c1a8b3c"
      url: "https://pub.dev"
    source: hosted
    version: "1.10.0"
  stack_trace:
    dependency: transitive
    description:
      name: stack_trace
      sha256: c3c7d8edb15bee7f0f74debd4b9c5f3c2ea86766fe4178eb2a18eb30a0bdaed5
      url: "https://pub.dev"
    source: hosted
    version: "1.11.0"
  stream_channel:
    dependency: transitive
    description:
      name: stream_channel
      sha256: "83615bee9045c1d322bbbd1ba209b7a749c2cbcdcb3fdd1df8eb488b3279c1c8"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.1"
  stream_transform:
    dependency: transitive
    description:
      name: stream_transform
      sha256: "14a00e794c7c11aa145a170587321aedce29769c08d7f58b1d141da75e3b1c6f"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.0"
  string_scanner:
    dependency: transitive
    description:
      name: string_scanner
      sha256: "556692adab6cfa87322a115640c11f13cb77b3f076ddcc5d6ae3c20242bedcde"
      url: "https://pub.dev"
    source: hosted
    version: "1.2.0"
  term_glyph:
    dependency: transitive
    description:
      name: term_glyph
      sha256: a29248a84fbb7c79282b40b8c72a1209db169a2e0542bce341da992fe1bc7e84
      url: "https://pub.dev"
    source: hosted
    version: "1.2.1"
  test_api:
    dependency: transitive
    description:
      name: test_api
      sha256: "75760ffd7786fffdfb9597c35c5b27eaeec82be8edfb6d71d32651128ed7aab8"
      url: "https://pub.dev"
    source: hosted
    version: "0.6.0"
  vector_math:
    dependency: transitive
    description:
      name: vector_math
      sha256: "80b3257d1492ce4d091729e3a67a60407d227c27241d6927be0130c98e741803"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.4"
  web:
    dependency: transitive
    description:
      name: web
      sha256: dc8ccd225a2005c1be616fe02951e2e342092edf968cf0844220383757ef8f10
      url: "https://pub.dev"
    source: hosted
    version: "0.1.4-beta"
sdks:
  dart: ">=3.1.1 <4.0.0"
  flutter: ">=3.13.0"

```

</details>

### Steps to reproduce

With the `camera` package (prior to the `camera_android_camerax` implementation), initializing the controller with `cameras[2]` would give access to the wide FoV camera.
In the new `camera_android_camerax` implementation, this does not work.

On a different device (Galaxy S22), setting zoom level to 0.6 worked.
However on this device (Galaxy XCover6 Pro) the minimum zoom is limited to 1.0.

So - on the Xcover6 Pro device, using the `camera_android_camerax` implementation, there is no way to get wide FoV images.

1. Initialize camera controller
2. Set zoom level to 0.6
3. Take picture

OR

1. Initialize camera controller with cameras[2]
2. Take picture

### Expected results

Get wide FoV image

### Actual results

Get normal FoV (zoom = 1.0) image

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'dart:io';

import 'package:flutter/material.dart';
import 'package:camera/camera.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});
  final String title;
  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;
  CameraController? controller;
  FileImage? fileImage;
  XFile? xFile;

  @override
  void initState() {
    super.initState();
    initCamera();
  }

  void initCamera() async {
    final cameras = await availableCameras();
    controller = CameraController(
        cameras[0],
        ResolutionPreset.ultraHigh,
        enableAudio: false,
        imageFormatGroup: ImageFormatGroup.jpeg
    );
    await controller!.initialize();
    await controller!.setFlashMode(FlashMode.off);

    await controller!.setZoomLevel(0.6);
    print("Minimum zoom level: ${await controller!.getMinZoomLevel()}");
  }

  void _incrementCounter() async {
    xFile = await controller!.takePicture();
    fileImage = FileImage(File(xFile!.path));
    setState(() {
      _counter++;
    });
  }

  SizedBox imageBox() {
    return
      SizedBox(
          height: 163,
          width: 210,
          child: Center(
              child: (fileImage != null) ? Image(
                  image: fileImage!,
                  height: 163,
                  width: 210,
                  fit: BoxFit.fitWidth
              ) : const Text('No image')
          )
      );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headlineMedium,
            ),
            imageBox(),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}


```

</details>

### Screenshots or Videos

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.13.3, on macOS 14.4.1 23E224 darwin-arm64, locale)
    • Flutter version 3.13.3 on channel stable at
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2524052335 (7 months ago), 2023-09-06 14:32:31 -0700
    • Engine revision b8d35810e9
    • Dart version 3.1.1
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.72.0)
    • VS Code at
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] VS Code (version 1.84.0)
    • VS Code at 
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • SM G736B (mobile) • android-arm64  • Android 14 (API 34)

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.

```

</details>

**Comment by huycozy** [comment]
Hi @livnehra 
>However on this device (Galaxy XCover6 Pro) the minimum zoom is limited to 1.0.

I think this is intended behavior since the supplied zoom value should be between 1.0 and the maximum supported zoom level (please see [this](https://github.com/flutter/packages/blob/caf48bc4cd97eec2a79200d80d8772e44944168a/packages/camera/camera_android_camerax/example/lib/camera_controller.dart#L610-L613))

>On a different device (Galaxy S22), setting zoom level to 0.6 worked.

Does this mean `getMinZoomLevel` will return 0.6 with `camerax`? (expected result) And this is a device-specific issue on `Galaxy XCover6 Pro`, right?

I checked this on my device `Realme 6, Android 11`, `getMinZoomLevel` returns 1.0 too.

**Comment by livnehra** [comment]
Yes:
1. On the XCover6 Pro, `getMinZoomLevel` returns 1.0, and that is what we can set.
2. On the S22 `getMinZoomLevel` returns 0.6 and that is what we can set.

So you could say that everything is working as intended.

But the point is that until now my app worked on XCover6 Pro by choosing `cameras[2]` - so I had a way to get the wide FoV. That method does not work anymore in the `camerax` implementation - there doesn't seem to be a difference between choosing `cameras[0]` or `cameras[2]` (the 2 back side cameras). Since the default Android implementation of `camera` is planned to be `camerax` starting next month - I will not be able to use it, since it offers no way to get the wide FoV on this model.

It should be noted that using the native camera app on the XCover6 pro allows zooming out all the way to 0.5 - so this is definitely a feature that the hardware supports, but does not seem to be supported by the `camerax` implementation.

For my purposes, a good enough solution would be to still allow choosing `cameras[2]` - I don't need access to the whole range of zooms between 0.5..1.0.

**Comment by huycozy** [comment]
Thanks for the update. I don't have such a device (XCover6 Pro) to check this. I will label this for other's input.

**Comment by reidbaker** [comment]
@camsim99 this is a feature that seems to be different between the 2 packages.

**Comment by camsim99** [comment]
> @camsim99 this is a feature that seems to be different between the 2 camera_android packages.

Yes this is true. My understanding from this issue, though, is that some devices still allow the wide angle and some do not using the CameraX plugin. This may be a result of some manufacturers not exposing it as described [in this discussion](https://groups.google.com/a/android.com/g/camerax-developers/c/fu3nSnk7Ox0), so we may need to try Camera2 interop to try exposing it for all devices that support it.

**Comment by reidbaker** [comment]
Based on that thread I would consider closing this as a device manufacture issue since the root of the problem seems to be manufactures providing a non deterministic way to expose wide angle as opposed to a bug in our (or camerax's) implementation)

**Comment by yaakovschectman** [comment]
From triage: We recognize that this is a feature regression. We will leave this open, but we believe the right place for this change is the CameraX library.

**Comment by ruicraveiro** [comment]
Hi, I am having the same behaviour on the following Samsung Galaxy devices:
- S23 SM-S911B/DS
- A23 SM-A236B/DSN
- XCover Pro SM-G715FN/DS

On all of them, when cycling through the available cameras, while using camera_android, I get wide FoV, but not while using camera_android_camerax.

**Comment by reidbaker** [comment]
@camsim99 do you have a link to the internal camerax bug tracking this?

**Comment by camsim99** [comment]
Based on https://b.corp.google.com/issues/247725855 it is supported.

> My understanding from this issue, though, is that some devices still allow the wide angle and some do not using the CameraX plugin. This may be a result of some manufacturers not exposing it as described [in this discussion](https://groups.google.com/a/android.com/g/camerax-developers/c/fu3nSnk7Ox0), so we may need to try Camera2 interop to try exposing it for all devices that support it.

I think we just believe that it may not be available for all devices based on that discussion.

**Comment by joseph-lewis** [comment]
Any update on this? My Galaxy S22 SE device also only has access to 4 cameras, none of them are the ultra-wide cameras

**Comment by ramaatamai** [comment]
Is it possible to get the ultra wide angle camera (0.5x) using  camera_android_camerax?

**Comment by ramaatamai** [comment]
@joseph-lewis Any update on this? My Galaxy S22 SE device also only has access to 4 cameras, none of them are the ultra-wide cameras

Did you find any solution?

**Comment by camsim99** [comment]
Unless I'm misunderstanding https://b.corp.google.com/issues/247725855 , `camera_android_camerax` is limited to supporting wide FOV to only the devices that the CameraX library can detect support it, and it's likely that devices seeing this issue are not recognizable by CameraX.

**Comment by camsim99** [comment]
Followed up with the CameraX team on this issue. What I learned:

**TLDR:** Either devices running into this don't support logical cameras, the camera selected does not represent the wide-angle lens, or something with the plugin camera selection via `availableCameras` is off (does not allow proper selection of all available logical cameras).

- To access wide FOV by setting < 1.0 zoom ratio, the device must support [logical cameras](https://developer.android.com/media/camera/camera2/multi-camera).
- It's possible that with camera_android (uses Camera2 versus CameraX) the wide FOV was previously accessible on some devices and not now because the angle was accessed as an individual camera
- Anyone running into this issue (or me if I can repro it) can check [CameraCharacteristics.LENS_INFO_AVAILABLE_FOCAL_LENGTHS](http://cameracharacteristics.lens_info_available_focal_lengths/) or [CameraInfo.getIntrinsicZoomRatio](http://camerainfo.getintrinsiczoomratio/) to verify if one logical camera is a wide-angle lens
  - https://groups.google.com/a/android.com/g/camerax-developers/c/tIaHMjtwdS4/m/Tco_yCqCAQAJ is an example of selecting wide angle lens
